tests/hpc/slurm_master.pm

```diff
@@ -19,38 +19,264 @@ use warnings;
 use testapi;
 use lockapi;
 use utils;
+use Data::Dumper;
+use Mojo::JSON;
+
+## TODO: provide better parser for HPC specific tests
+sub validate_result {
+    my ($result) = @_;
+
+    if ($result == 0) {
+        return 'PASS';
+    } elsif ($result == 1) {
```

Comment:
just as naive question, why don't we just assume everything !=0 is 'FAIL' ? Why do you need undef situations?

Reply:
simply because of TODO there: "## TODO: provide better parser for HPC specific tests"

Current approach is obviously naive and simplistic. Overall I hope you can see my direction:
abstract cluster provisioning and separate tests to the outside files. But yeah I would still do it in steps, as I'm really busy with tons of tasks and can't really spend long time into inventing hpc runner now.

I assume I eventually will end up having more values than TRUE, FALSE so that's why it looks like it looks ;-)

wdyt?